src/store/storage/index.ts

```diff
@@ -4,33 +4,35 @@ import { Action } from '@ngrx/store';
 
 import { Keymap } from '../../config-serializer/config-items/Keymap';
 import { Macro } from '../../config-serializer/config-items/Macro';
+import { UhkConfiguration } from '../../config-serializer/config-items/UhkConfiguration';
+
 import { KeymapActions, MacroActions } from '../actions';
 import { AppState } from '../index';
 import { Electron } from './electron';
 import { Local } from './local';
 
-import { UhkConfiguration } from '../../config-serializer/config-items/UhkConfiguration';
-import { DataProviderService } from '../../services/data-provider.service';
-
 @Injectable()
 export class DataStorage {
 
     private _environment: Local | Electron;
+    private uhkConfiguration: UhkConfiguration;
+    private uhkPresets: Keymap[];
 
-    constructor(private dataProvider: DataProviderService) {
+    constructor() {
+        this.initUHKJson();
         this.detectEnvironment();
     }
 
     initialState(): AppState {
-        let config: UhkConfiguration = this._environment.getConfig() || this.dataProvider.getUHKConfig();
+        let config: UhkConfiguration = this._environment.getConfig() || this.uhkConfiguration;
```

Comment:
But this is called only once. So if a user deletes the local storage while using agent we have a problem, because this `initialState `will not be called again.